extractor/filesystem/language/cpp/conanlock/extractor.go

```diff
@@ -0,0 +1,240 @@
+// Package conanlock extracts conan.lock files.
+package conanlock
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/osv"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/google/osv-scalibr/purl"
+)
+
+type conanReference struct {
+	Name            string
+	Version         string
+	Username        string
+	Channel         string
+	RecipeRevision  string
+	PackageID       string
+	PackageRevision string
+	TimeStamp       string
+}
+
+type conanGraphNode struct {
+	Pref      string `json:"pref"`
+	Ref       string `json:"ref"`
+	Options   string `json:"options"`
+	PackageID string `json:"package_id"`
+	Prev      string `json:"prev"`
+	Path      string `json:"path"`
+	Context   string `json:"context"`
+}
+
+type conanGraphLock struct {
+	Nodes map[string]conanGraphNode `json:"nodes"`
+}
+
+type conanLockFile struct {
+	Version string `json:"version"`
+	// conan v0.4- lockfiles use "graph_lock", "profile_host" and "profile_build"
+	GraphLock    conanGraphLock `json:"graph_lock,omitempty"`
+	ProfileHost  string         `json:"profile_host,omitempty"`
+	ProfileBuild string         `json:"profile_build,omitempty"`
+	// conan v0.5+ lockfiles use "requires", "build_requires" and "python_requires"
+	Requires       []string `json:"requires,omitempty"`
+	BuildRequires  []string `json:"build_requires,omitempty"`
+	PythonRequires []string `json:"python_requires,omitempty"`
+}
+
+// TODO this is tentative and subject to change depending on the OSV schema
```

Comment:
Add a bug for TODOs or don't prefix comment with a TODO

Reply:
Removed, I think this is now in the osv-schema, so no need for this comment.